sympy/polys/matrices/sdm.py

```diff
@@ -970,3 +1017,22 @@ def sdm_particular_from_rref(A, ncols, pivots):
         if Ain is not None:
             P[j] = Ain / A[i][j]
     return P
+
+
+def sdm_applyfunc(M, shape, f, domain):
+    if not f(domain.zero):
+        return unop_dict(M, f)
+
+    nrows, ncols = shape
+    fzero = f(domain.zero)
```

Comment:
`f(domain.zero)` is computed above. The result should be reused rather than calling `f` again.